internal/restic/node_test.go

```diff
@@ -382,3 +383,14 @@ func TestSymlinkSerializationFormat(t *testing.T) {
 		test.Assert(t, n2.LinkTargetRaw == nil, "quoted link target is just a helper field and must be unset after decoding")
 	}
 }
+
+func TestNodeRestoreMetadataError(t *testing.T) {
```

Comment:
I would have added more interesting tests here, but mocking sys calls is a little awkward in Go, I found.